src/srs-filter.c

```diff
@@ -273,14 +319,18 @@ xxfi_srs_milter_envfrom(SMFICTX* ctx, char** argv) {
   cd->recip_remote = 0;
 
   // strore MAIL FROM: address
-  cd->sender = (char *) malloc(strlen(argv[0])-1);
+  cd->sender = (char *) malloc(strlen(argv[0])+1);
   if (!cd->sender) {
     // memory allocation problem
     cd->state |= SS_STATE_INVALID_MSG;
     return SMFIS_CONTINUE;
   }
-  strncpy(cd->sender, argv[0]+1, strlen(argv[0])-2);
-  cd->sender[strlen(argv[0])-2] = '\0';
+  if (argv[0][0] == '<') {
+    strncpy(cd->sender, argv[0]+1, strlen(argv[0])-2);
+    cd->sender[strlen(argv[0])-2] = '\0';
```

Comment:
strnlen preferred.

also, for both of these lines, can you please add a comment for readability? I recognize this isn't new code, just refactoring, but honestly I can't follow what this does (easily), i.e. references to +1 and -2 offsets.

Do we need to check that argv[0] isn't non-null before we dereference to [0][0]? why are we looking for a `<` in argv[0][0] anyhow? isn't that the first character used to invoke our command name?